movement/watch_faces/complication/totp_face.c

```diff
@@ -75,15 +219,11 @@ bool totp_face_loop(movement_event_t event, movement_settings_t *settings, void
             movement_move_to_face(0);
             break;
         case EVENT_ALARM_BUTTON_UP:
-            if (index + 1 < num_keys) {
-                totp_state->current_key_offset += key_sizes[index];
-                totp_state->current_index++;
-            } else {
-                // wrap around to first key
-                totp_state->current_key_offset = 0;
-                totp_state->current_index = 0;
+            if (num_totp_records == 0) {
+                break;
             }
-            TOTP(keys + totp_state->current_key_offset, key_sizes[totp_state->current_index], timesteps[totp_state->current_index]);
+            totp_state->current_index = (index + 1) % num_totp_records;
+            TOTP(totp_records[totp_state->current_index].secret, totp_records[totp_state->current_index].secret_size, totp_records[totp_state->current_index].period);
```

Comment:
TODO: line looks a bit long. Probably should just set 'index' as above. Hmm, and feels like I should reset steps to force getCodeFromTimestamp to happen again.